test/src/main/java/org/axonframework/test/fixture/AxonTestFixture.java

```diff
@@ -0,0 +1,594 @@
+        this.customization = customization.apply(new Customization());
+        this.configuration = configuration;
+        this.messageTypeResolver = configuration.getComponent(MessageTypeResolver.class);
+        this.commandBus = (RecordingCommandBus) configuration.getComponent(CommandBus.class);
+        this.eventSink = (RecordingEventSink) configuration.getComponent(EventSink.class);
+    }
+
+    public static AxonTestPhase.Setup with(ApplicationConfigurer<?> configurer) {
+        return with(configurer, c -> c);
+    }
+
+    public static AxonTestPhase.Setup with(ApplicationConfigurer<?> configurer,
+                                           UnaryOperator<Customization> customization) {
+        var testConfigurer = new TestApplicationConfigurer(configurer);
+        var configuration = testConfigurer.build();
+        return with(configuration, customization);
+    }
+
+    public static AxonTestPhase.Setup with(TestApplicationConfigurer configurer) {
+        var configuration = configurer.build();
+        return with(configuration, c -> c);
+    }
+
+    public static AxonTestPhase.Setup with(TestApplicationConfigurer configurer,
+                                           UnaryOperator<Customization> customization) {
+        var configuration = configurer.build();
+        return with(configuration, customization);
+    }
+
+    public static AxonTestPhase.Setup with(NewConfiguration configuration) {
+        return new AxonTestFixture(configuration, c -> c);
+    }
+
+    public static AxonTestPhase.Setup with(NewConfiguration configuration, UnaryOperator<Customization> customization) {
+        return new AxonTestFixture(configuration, customization);
+    }
+
+    @Override
+    public AxonTestPhase.Given given() {
+        return new Given(configuration, customization, commandBus, eventSink, messageTypeResolver);
+    }
+
+    @Override
+    public AxonTestPhase.When when() {
+        return new When(configuration, customization, messageTypeResolver, commandBus, eventSink);
+    }
+
+    public record Customization(List<FieldFilter> fieldFilters) {
+
+        public Customization() {
+            this(new ArrayList<>());
+        }
+
+        public Customization registerFieldFilter(FieldFilter fieldFilter) {
+            this.fieldFilters.add(fieldFilter);
+            return this;
+        }
+
+        public Customization registerIgnoredField(Class<?> declaringClass, String fieldName) {
+            return registerFieldFilter(new IgnoreField(declaringClass, fieldName));
+        }
+    }
+
+    /**
+     * Every given method invocation spawn new unit of work.
+     */
+    static class Given implements AxonTestPhase.Given {
+
+        private final NewConfiguration configuration;
+        private final Customization customization;
+        private final RecordingCommandBus commandBus;
+        private final RecordingEventSink eventSink;
+        private final MessageTypeResolver messageTypeResolver;
```

Comment:
Nit: Perhaps you prefer this style, but I am wondering. If the `Given` is an inner-class, why not let it access the variables from the parent class? I do not think they differ, right?

Reply:
Or, you make this a different, package private, class, of course! I honestly think that wouldn't be to bad either, as this class will definitely get larger in the long run.

Reply:
Smart!